python/cudf/cudf/_libxx/join.pyx

```diff
@@ -16,10 +16,6 @@ cpdef join(lhs, rhs, left_on, right_on, how, method, left_index=False, right_ind
     cdef vector[int] right_on_ind
     cdef vector[pair[int, int]] columns_in_common
 
-    # convert both left and right to c table objects
-    cdef Table c_lhs = lhs
-    cdef Table c_rhs = rhs
```

Comment:
I think you still want these as calling this from Python won't automatically turn them into c objects